common/src/main/java/io/druid/math/expr/Evals.java

```diff
@@ -40,4 +46,39 @@ public static Number toNumber(Object value)
     }
     return longValue;
   }
+
+  public static boolean isConstant(Expr expr)
+  {
+    return expr instanceof DoubleExpr || expr instanceof LongExpr || expr instanceof StringExpr;
```

Comment:
Is it feasible to add `isConstant` as a method on `Expr`?

I think it's nice to have all the logic for a particular expr together in the expr class, rather than spread around the project in methods like `isConstant` and `flatten`. That makes it easier to add new exprs without having to update a lot of different places in the code.